src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -355,6 +364,118 @@ nlohmann::json SysInfo::getNetworks() const
     return networks;
 }
 
+std::pair<std::string, std::string> parseProcessInfo(char path[])
+{
+    // Open stat file
+    std::pair<std::string, std::string> processInfo;
+    std::ifstream file(path, std::ios::in | std::ios::binary);
+    std::string statContent;
+
+    // Get stat file content
+    std::getline(file, statContent);
+
+    // Parse PID and process name
+    int openParentesysPos = std::string(statContent).find("(");
+    int closeParentesysPos = std::string(statContent).find(")");
+    int spacePos = std::string(statContent).find(" ");
+
+    processInfo.first = std::string(statContent).substr(0, spacePos);
+    processInfo.second = std::string(statContent).substr(openParentesysPos + 1, closeParentesysPos - openParentesysPos - 1);
+
+    file.close();
+
+    return processInfo;
+}
+
+void findInodeMatch(char path[], std::vector<int64_t>& inodes, bool& matchInode, const nlohmann::json& JSONInput)
+{
+    // Read symbolic link files
+    char fdFileContent[KByte] {0};
+
+    if (readlink(path, fdFileContent, sizeof(fdFileContent)) > 0)
+    {
+        // Parse inode from symbolic link.
+        int openBracketPos = std::string(fdFileContent).find("[");
+        int closeBracketPos = std::string(fdFileContent).find("]");
+        std::string match = std::string(fdFileContent).substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1);
+        bool exists{false};
```

Comment:
You could avoid creating this variable by doing the following:

```c++
    // Read symbolic link files
    char fdFileContent[8] {0};

    if (readlink(path, fdFileContent, sizeof(fdFileContent)) > 0)
    {
        // Parse inode from symbolic link.
        const int openBracketPos = std::string(fdFileContent).find("[");
        const int closeBracketPos = std::string(fdFileContent).find("]");
        const std::string match = std::string(fdFileContent).substr(openBracketPos + 1, closeBracketPos - openBracketPos - 1);
        int64_t matchN{0};

        for (const auto& i : JSONInput)
        {
            std::stringstream ss;
            ss << match;
            ss >> matchN;

            std::cout << ss.str()  << std::endl;
            if (i.at("inode") == matchN)
            {
                matchInode = true;
                inodes.push_back(matchN);
            }
        }
    }
```

Reply:
Changed